commands/ls_print.go

```diff
@@ -0,0 +1,206 @@
+package commands
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/docker/buildx/builder"
+	"github.com/docker/buildx/store"
+	"github.com/docker/buildx/util/platformutil"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/formatter"
+)
+
+const (
+	lsNameNodeHeader       = "NAME/NODE"
+	lsDriverEndpointHeader = "DRIVER/ENDPOINT"
+	lsStatusHeader         = "STATUS"
+	lsLastActivityHeader   = "LAST ACTIVITY"
+	lsBuildkitHeader       = "BUILDKIT"
+	lsPlatformsHeader      = "PLATFORMS"
+
+	lsIndent = ` \_ `
+
+	lsDefaultTableFormat = "table {{.NameNode}}\t{{.DriverEndpoint}}\t{{.Status}}\t{{.Buildkit}}\t{{.Platforms}}"
+	lsDefaultRawFormat   = `{{- if not .IsNode -}}
+name: {{.NameNode}}
+{{- if .DriverEndpoint }}
+driver: {{.DriverEndpoint}}
+{{- end }}
+{{- if .LastActivity }}
+last_activity: {{.LastActivity}}
+{{- end }}
+{{- if .Error }}
+error: {{.Error}}
+{{- else }}
+nodes:
+{{- end }}
+{{- else }}  name: {{.NameNode}}
+  endpoint: {{.DriverEndpoint}}
+  status: {{.Status}}
+  {{- if .Buildkit }}
+  buildkit: {{.Buildkit}}
+  {{- end }}
+  {{- if .Platforms }}
+  platforms: {{.Platforms}}
+  {{- end }}
+  {{- if .Error }}
+  error: {{.Error}}
+  {{- end }}
+{{- end }}
+`
+)
+
+func lsPrint(dockerCli command.Cli, current *store.NodeGroup, builders []*builder.Builder, format string) (hasErrors bool, _ error) {
+	switch format {
+	case formatter.TableFormatKey:
+		format = lsDefaultTableFormat
```

Comment:
We may have been inconsistent with this; slightly wondering if we should consider the default to be a `table` or not. (because of the nested presentation). We probably called it "table" for `docker service ps` (or which one was it?)

The default presentation _COULD_ be considered more of a `tree` format (https://github.com/moby/moby/issues/44582#issuecomment-1456938816), which could keep the road open to implement a "flat" table presentation.

(I haven't fully thought this through though!) /cc @jalonsogo

Reply:
That sounds like major change that should be first done upstream and then tackled in cli plugins no?